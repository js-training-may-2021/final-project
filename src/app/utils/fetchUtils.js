import axios from 'axios';
import _ from 'lodash';

const isDev = process.env.NODE_ENV === 'development';
const pathApi = isDev ? 'http://localhost:3000' : '/api/v1';
export const limit = 20;

const fetchPokemons = (page, query, attr, order) => axios.get(`${pathApi}/pokemons?q=${query}&_page=${page}&_limit=${limit}&_sort=${attr}&_order=${order}`);
const normalizedQuery = (query) => _.trimStart(query, '#');

export const mappingFetchPokemons = {
  idAsc: (page, query) => fetchPokemons(page, normalizedQuery(query), 'id', 'asc'),
  idDesc: (page, query) => fetchPokemons(page, normalizedQuery(query), 'id', 'desc'),
  nameAsc: (page, query) => fetchPokemons(page, normalizedQuery(query), 'name', 'asc'),
  nameDesc: (page, query) => fetchPokemons(page, normalizedQuery(query), 'name', 'desc'),
};

export const fetchPokemonById = (id) => axios.get(`${pathApi}/pokemons?id=${id}`);

export const fetchData = () => (
  axios({
    method: 'post',
    url: '/api/v1/data',
    data: { token: localStorage.token },
    timeout: 4000,
  })
);

export const fetchCatch = (pokemon) => axios({
  method: 'post',
  url: '/api/v1/catchPokemon',
  data: { ...pokemon, token: localStorage.token },
  timeout: 4000,
});

export const fetchLogin = (data) => axios({
  method: 'post',
  url: '/api/v1/login',
  data,
  timeout: 4000,
});
