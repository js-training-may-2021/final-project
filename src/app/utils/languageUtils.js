import { useTranslation } from 'react-i18next';

export const setDefaultLanguage = () => {
  const defaultLanguage = 'ru';

  if (localStorage.getItem('language') === null) {
    localStorage.setItem('language', defaultLanguage);
  }
};

export const getCurrentLanguage = () => localStorage.getItem('language');

export const toggleLanguage = () => {
  const currentLanguage = localStorage.getItem('language');
  const newLanguage = currentLanguage === 'ru' ? 'en' : 'ru';
  localStorage.setItem('language', newLanguage);

  return newLanguage;
};

export const loc = (text) => {
  const { t } = useTranslation();
  return t(text);
};
