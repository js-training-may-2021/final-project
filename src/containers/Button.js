import React from 'react';
import styled from 'styled-components';
import { useDispatch } from 'react-redux';
import { catchPokemon } from '../store/actions';

const Btn = styled.button`
    border: none;
    border-radius: ${props => props.radius ? "16px" : "0px"};
    border-top-left-radius: 0px;
    border-top-right-radius:  0px;
    margin-bottom: 0;
    width: 100%;
    padding: 15px 0;
    background-color: #FFE34E;
    font-size: 14px;
    cursor: pointer;

    &:disabled {
        background-color: #DFDFDF;
        color: #fff;
    }

    &:hover {
        background-color: #FFEF9C;
    }
`;

const Button = ({disabled, id, name, isCaught}) => {

    const dispatch = useDispatch();

    const clickHandler = (id, name) => (e) => {
        e.preventDefault();
        dispatch(catchPokemon({id, name}))
    }

    return (
        <Btn disabled={disabled} onClick={clickHandler(id, name)}>
            { isCaught ? 'Caught!' : 'Catch!'}
        </Btn>
    )
}

export default Button;