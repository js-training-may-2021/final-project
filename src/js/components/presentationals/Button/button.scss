@import '../../../../styles/variables';

.button {
  height: 60px;
  width: 100%;
  border: none;
  border-radius: 10px;
  background: map-get($colorScheme, "button");
  color: map-get($colorScheme, "textSecondary");
  font-size: 1rem;
  font-weight: 700;

  &:hover,
  &:active,
  &:focus {
    background: transparent;
    border: 2px solid map-get($colorScheme, "button");
    color: map-get($colorScheme, "button");
  }
}