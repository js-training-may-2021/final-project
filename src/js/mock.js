export const pokemons = [
  {
    name: 'bulbasaur',
    id: 1,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'ivysaur',
    id: 2,
    isCaught: true,
    catchDate: '23 June 2021',
  },
  {
    name: 'venusaur',
    id: 3,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'charmander',
    id: 4,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'charmeleon',
    id: 5,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'charizard',
    id: 6,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'squirtle',
    id: 7,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'wartortle',
    id: 8,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'blastoise',
    id: 9,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'caterpie',
    id: 10,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'metapod',
    id: 11,
    isCaught: false,
    catchDate: null,
  }];  

export const uncaughtPokemon = pokemons[0];
export const caughtPokemon = pokemons[1];
