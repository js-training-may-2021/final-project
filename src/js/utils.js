import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';

export const POKEMONS_PER_PAGE = 10;

let currentPage = 1;

export const getCurrentPage = () => {
	return currentPage++;
};

export const AppRoute = {
  MAIN: '/',
  DETAIL: '/detail/:id',
  CAUGHT: '/caught'
};

const monthNames = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

export const getCatchDate = () => {
  const date = new Date();

  const day = date.getDate();
  const month = monthNames[date.getMonth()];
  const year = date.getFullYear();

  return `${day} ${month} ${year}`
};

export const noop = () => {};
export const history = createMemoryHistory();
const mockStore = configureStore();
export const testStore = mockStore({});
