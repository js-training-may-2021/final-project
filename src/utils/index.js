export const firstLetterToUpperCase = (str) => {
  if (str === undefined) {
    return str;
  } else {
    return str
      .toLowerCase()
      .split(" ")
      .map((s) => s.charAt(0).toUpperCase() + s.substring(1))
      .join(" ");
  }
};

export const formatDate = (dateInput) => {
  return dateInput.toLocaleString("ru-RU");
};

export const padToThree = (number) => number <= 999 ? `00${number}`.slice(-3) : number;

export const handleErrors = (response) => {
  if (!response.ok) {
    throw Error(`${response.status}: ${response.statusText}`);
  }
  return response;
};
