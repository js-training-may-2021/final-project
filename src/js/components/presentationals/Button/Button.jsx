import React from 'react';
import PropTypes from 'prop-types';
import './button.scss';

const Button = (props) => {
  const {
    isToggleChecked,
    onClick,
    onToggleClick,
  } = props;

	return (
		<button
			className={"card__button button" + (isToggleChecked ? " button--checked" : "")}
			type="button"
      onClick={() => {
        onClick();
        onToggleClick();
      }}
		>
			{isToggleChecked ? `Release` : `Catch!`}
		</button>
	);
};

Button.propTypes = {
  isToggleChecked: PropTypes.bool.isRequired,
	onClick: PropTypes.func,
  onToggleClick: PropTypes.func,
}

export default React.memo(Button);
