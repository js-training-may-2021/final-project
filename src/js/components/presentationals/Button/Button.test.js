import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button.jsx';
import {noop} from '../../../utils.js';

it(`Render Button of uncaught pokemon`, () => {
  const tree = renderer
    .create(
      <Button
        isToggleChecked={false}
        onClick={noop}
        onToggleClick={noop}
      />)
    .toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree.children[0]).toBe('Catch!');
  expect(tree.props.className).toBe('card__button button');
});

it(`Render Button of caught pokemon`, () => {
  const tree = renderer
    .create(
      <Button
        isToggleChecked={true}
        onClick={noop}
        onToggleClick={noop}
      />)
    .toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree.children[0]).toBe('Release');
  expect(tree.props.className).toBe('card__button button button--checked');
});
