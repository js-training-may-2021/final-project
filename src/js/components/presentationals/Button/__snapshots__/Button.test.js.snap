// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button of caught pokemon 1`] = `
<button
  className="card__button button button--checked"
  onClick={[Function]}
  type="button"
>
  Release
</button>
`;

exports[`Render Button of uncaught pokemon 1`] = `
<button
  className="card__button button"
  onClick={[Function]}
  type="button"
>
  Catch!
</button>
`;
